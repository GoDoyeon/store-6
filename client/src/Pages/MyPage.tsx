import React, { useState } from 'react';
import styled from 'styled-components';

const MyPageWrapper = styled.div`
  width: 100%;
  display:flex;
  flex-direction:column;
  align-items:center;
`

const MyPageTitleContainer = styled.div`
  margin-bottom: 90px;
`

const MyPageTitle = styled.div`
  font-size: 25px;
  padding: 6px 40px 7px 10px;
`

const MyPageRowContainer = styled.div`
  display:flex;
  align-items:center;
  justify-content:space-around;
  width:630px;
  margin : 30px 0;
`

const MyPageRowTitle = styled.div`
  font-size:20px;
`

const MyPagePhotoContentContainer = styled.div`

`

const MyPageEmailContentContainer = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  top : 6px;
`

const MyPageEmailInput = styled.input`
  border: none;
  outline: none;
  margin-bottom: 15px;
  padding-left:10px;
  font-size: 14px;
`

const MyPageButton = styled.div`
  border: none;
  outline: none;
  font-size1: 15px;
`

const MyPagePhoneContentContainer = styled.div`
  display:flex;
  position: relative;
  top: 3px;
`
const MyPagePhoneInputContainer = styled.div`
  width:100px;
`


const MyPagePhoneInput = styled.input`
  border: none;
  outline: none;
  width:90px;
  font-size: 14px;
  text-align: center;
`


const DivideContainer = styled.div`
  width: 30px;
  text-align: center;
  position: relative;
  top: 6px;
  right: 2px;
`

const MyPage: React.FC = () => {
  const [profile, setProfile] = useState<string>('');
  const [email, setEmail] = useState<string>('');
  const [firstPhoneValue, setFirstPhoneValue] = useState<string>('');
  const [middlePhoneValue, setMiddlePhoneValue] = useState<string>('');
  const [lastPhoneValue, setLastPhoneValue] = useState<string>('');

  return (
    <MyPageWrapper>
      <MyPageTitleContainer>
        <MyPageTitle>마이페이지</MyPageTitle>
        <img src="https://user-images.githubusercontent.com/47776356/129517220-96148b08-55d7-46fb-adaa-1c04152e9e09.png"/>
      </MyPageTitleContainer>
      <div>
        <MyPageRowContainer>
          <MyPageRowTitle>사진</MyPageRowTitle>
          <MyPagePhotoContentContainer>
            <div>
              <label htmlFor="img" className="input-preview"></label>
              <input name="img" id="img" className="input-preview__src" type="file"/>
            </div>
            <div>사진을 클릭하면 등록된 사진을 수정할 수 있습니다.</div>
          </MyPagePhotoContentContainer>
          <div></div>
        </MyPageRowContainer>
        <img src="https://user-images.githubusercontent.com/47776356/129521962-9d063baa-2794-4936-9635-ba989f1c831f.png"/>
        <MyPageRowContainer>
          <MyPageRowTitle>이메일</MyPageRowTitle>
          <MyPageEmailContentContainer>
            <MyPageEmailInput/>
            <img src="https://user-images.githubusercontent.com/47776356/129522373-57c5806a-1959-47f2-9bb0-4f189f6ed6cf.png"/>
          </MyPageEmailContentContainer>  
          <MyPageButton>수정</MyPageButton>
        </MyPageRowContainer>
        <img src="https://user-images.githubusercontent.com/47776356/129521962-9d063baa-2794-4936-9635-ba989f1c831f.png"/>
        <MyPageRowContainer>
          <MyPageRowTitle>연락처</MyPageRowTitle>
          <MyPagePhoneContentContainer>
            <MyPagePhoneInputContainer>
              <MyPagePhoneInput />
              <img src="https://user-images.githubusercontent.com/47776356/129524612-34323042-93bf-49be-b346-5d1e9433d614.png"/>
            </MyPagePhoneInputContainer>
            <DivideContainer>
              <img src="https://user-images.githubusercontent.com/47776356/129524462-7931a7ae-73d2-4b7e-a911-01c4f780a99a.png"/>
            </DivideContainer>
            <MyPagePhoneInputContainer>
              <MyPagePhoneInput/>
              <img src="https://user-images.githubusercontent.com/47776356/129524612-34323042-93bf-49be-b346-5d1e9433d614.png"/>
            </MyPagePhoneInputContainer>
            <DivideContainer>
              <img src="https://user-images.githubusercontent.com/47776356/129524462-7931a7ae-73d2-4b7e-a911-01c4f780a99a.png"/>
            </DivideContainer>
            <MyPagePhoneInputContainer>
              <MyPagePhoneInput/>
              <img src="https://user-images.githubusercontent.com/47776356/129524612-34323042-93bf-49be-b346-5d1e9433d614.png"/>
            </MyPagePhoneInputContainer>
          </MyPagePhoneContentContainer>
          <MyPageButton>완료</MyPageButton>
        </MyPageRowContainer>
      </div>
    </MyPageWrapper>
  );
};

export default MyPage;
