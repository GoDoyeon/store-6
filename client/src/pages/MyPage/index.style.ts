import styled from 'styled-components';

export const RowWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 630px;
  height: 70px;
  margin: 30px 0;
`;

export const RowTitle = styled.div`
  font-size: 20px;
`;

export const PhotoWrapper = styled.div`
  display: flex;
  align-items: center;
`;

export const Button = styled.div`
  border: none;
  outline: none;
  font-size: 15px;
`;

export const ImagePreview = styled.label`
  border-radius: 50%;
  width: 60px;
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  transition: ease-in-out 750ms;
  margin-right: 40px;
  box-shadow: rgba(50, 50, 93, 0.25) 0px 2px 5px -1px,
    rgba(0, 0, 0, 0.3) 0px 1px 3px -1px;
  cursor: pointer;
`;

export const EmailWrapper = styled.div`
  position: relative;
  bottom: 4px;
`;

export const ImageDesc = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;

  p {
    font-size: 15px;
  }
`;

export const ImageInput = styled.input`
  display: none;
`;

export const MyPageContent = styled.div`
  margin: 0 auto;
  padding-top: 70px;
`;
