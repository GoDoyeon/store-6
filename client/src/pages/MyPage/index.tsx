import React, { useEffect, useContext } from 'react';

import Divider from '~/components/common/Divider';
import UserInfoInput from '~/components/my/UserInfoInput';
import PhoneInput from '~/components/my/PhoneInput';
import EmailInput from '~/components/my/EmailInput';
import SubPageWrapper from '~/components/subpage/SubPageWrapper';
import SubPageHeader from '~/components/subpage/SubPageHeader';
import SubPageHeaderItem from '~/components/subpage/SubPageHeaderItem';

import { setUserInfo } from '~/stores/userModule';
import { putMe } from '~/lib/api/users';
import UserContext from '~/lib/contexts/userContext';
import { alert } from '~/utils/modal';
import { REG_EMAIL, REG_IMAGE, REG_PHONE } from '~/utils/validation';

import {
  RowWrapper,
  RowTitle,
  PhotoWrapper,
  ImagePreview,
  ImageDesc,
  ImageInput,
  MyPageContent,
} from './index.style';
import { useHistory } from '~/core/Router';

const message = {
  emailUpdateSuccess: '이메일이 수정되었습니다.',
  phoneUpdateSuccess: '연락처가 수정되었습니다.',
  photoUpdateSuccess: '사진이 수정되었습니다.',
  notCorrectImage: '올바른 이미지파일 형식이 아닙니다.',
  emailUpdateFail: '이메일 수정에 실패했습니다.',
  emailInvalid: '올바른 이메일 형식이 아닙니다.',
  phoneInvalid: '올바른 연락처 형식이 아닙니다.',
  phoneUpdateFail: '연락처 수정에 실패했습니다.',
  photoUpdateFail: '사진 수정에 실패했습니다',
  photoUpdateLarge: '1mb 이하의 사진만 업로드 가능합니다.',
};

const MyPage: React.FC = () => {
  const { push } = useHistory();
  const { user: userState, userDispatch } = useContext(UserContext);
  const mb = 1;

  const handleSubmitEmail = (value: string) => {
    return putMe({ email: value })
      .then(() => {
        alert(message.emailUpdateSuccess);
        userDispatch(setUserInfo({ email: value }));
      })
      .catch(() => alert(message.emailUpdateFail));
  };

  const handleSubmitPhone = (value: string) => {
    return putMe({ phone: value })
      .then(() => {
        alert(message.phoneUpdateSuccess);
        userDispatch(setUserInfo({ phone: value }));
      })
      .catch(() => alert(message.phoneUpdateFail));
  };

  const handleSubmitProfile = (file: File, fileToString: string) => {
    return putMe({ profile: file })
      .then(() => {
        userDispatch(setUserInfo({ ...userState.user, profile: fileToString }));
        alert(message.photoUpdateSuccess);
      })
      .catch(() => alert(message.photoUpdateFail));
  };

  const handleImageInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    const file = e.target.files[0];
    if (!file) {
      return;
    }
    if (!file.type.match(REG_IMAGE)) {
      e.target.value = '';
      alert(message.notCorrectImage);
      return;
    }
    const reader = new FileReader();
    reader.readAsDataURL(e.target.files[0]);
    reader.onload = (event) => {
      if (file.size / (1024 * 1024) > mb) {
        alert(message.photoUpdateLarge);
      } else {
        handleSubmitProfile(file, event.target.result.toString());
      }
    };
  };

  const emailValidator = (value: string) => {
    return REG_EMAIL.test(value);
  };
  const phoneValidator = (value: string) => {
    return REG_PHONE.test(value);
  };

  useEffect(() => {
    if (!userState.isLoggedIn) {
      push('/', { from: '/me', error: 'accessWithoutToken' });
    }
  }, [userState]);

  return (
    <SubPageWrapper width="700px">
      <SubPageHeader>
        <SubPageHeaderItem>마이페이지</SubPageHeaderItem>
      </SubPageHeader>
      <MyPageContent>
        <RowWrapper>
          <RowTitle>사진</RowTitle>
          <PhotoWrapper>
            <div>
              <ImagePreview image={userState.user?.profile} size="60px" />
              <ImageInput
                onChange={handleImageInput}
                accept=".jpg, .png, .jpeg"
                id="img"
                type="file"
              />
            </div>
            <ImageDesc>
              <p>사진을 클릭하면 등록된 사진을 수정할 수 있습니다.</p>
            </ImageDesc>
          </PhotoWrapper>
          <div />
        </RowWrapper>

        <Divider width="630px" direction="horizontal" />

        <UserInfoInput
          title="이메일"
          showWarning={() => alert(message.emailInvalid)}
          value={userState.user?.email}
          placeholder={userState.user?.email}
          onSubmit={handleSubmitEmail}
          inputComponent={EmailInput}
          validator={emailValidator}
        />

        <Divider width="630px" direction="horizontal" />

        <UserInfoInput
          title="연락처"
          showWarning={() => alert(message.phoneInvalid)}
          value={userState.user?.phone}
          placeholder={userState.user?.phone?.split('-').join(' ')}
          onSubmit={handleSubmitPhone}
          inputComponent={PhoneInput}
          validator={phoneValidator}
        />
      </MyPageContent>
    </SubPageWrapper>
  );
};

export default MyPage;
