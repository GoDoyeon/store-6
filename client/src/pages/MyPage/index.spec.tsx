import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import MyPage from './index';

describe('<MyPage />', () => {
  it('should render same with snapshot', () => {
    const { container } = render(<MyPage />);
    expect(container).toMatchSnapshot();
  });
});
