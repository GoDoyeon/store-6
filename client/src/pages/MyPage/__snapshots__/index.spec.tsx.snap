// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyPage /> should render same with snapshot 1`] = `
.c2 {
  width: 700px;
  height: 4px;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  background: url(https://store-6-bucket.s3.ap-northeast-2.amazonaws.com/common/line.svg) bottom left no-repeat;
  background-size: cover;
}

.c9 {
  width: 630px;
  height: 1px;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  background: url(https://store-6-bucket.s3.ap-northeast-2.amazonaws.com/common/line.svg) bottom left no-repeat;
  background-size: cover;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  width: 630px;
  height: 70px;
  margin: 30px 0;
}

.c11 {
  font-size: 20px;
}

.c13 {
  border: none;
  outline: none;
  font-size: 15px;
  cursor: pointer;
  color: #000;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  bottom: 4px;
}

.c15 {
  text-align: center;
  box-sizing: border-box;
  width: 100px;
  height: 34px;
  padding: 0 16px 10px 16px;
  margin-top: 16px;
  line-height: 25px;
  font-size: 14px;
  background: bottom left / contain no-repeat url();
}

.c15::-webkit-input-placeholder {
  color: #999;
}

.c15::-moz-placeholder {
  color: #999;
}

.c15:-ms-input-placeholder {
  color: #999;
}

.c15::placeholder {
  color: #999;
}

.c12 {
  box-sizing: border-box;
  width: 370px;
  height: 34px;
  padding: 0 16px 10px 16px;
  margin-top: 16px;
  line-height: 25px;
  font-size: 20px;
  background: bottom left / contain no-repeat url();
}

.c12::-webkit-input-placeholder {
  color: #999;
}

.c12::-moz-placeholder {
  color: #999;
}

.c12:-ms-input-placeholder {
  color: #999;
}

.c12::placeholder {
  color: #999;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 70px;
  margin-top: 90px;
}

.c1 {
  font-size: 25px;
  padding: 6px 40px 20px 30px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  width: 630px;
  height: 70px;
  margin: 30px 0;
}

.c4 {
  font-size: 20px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  border-radius: 50%;
  width: 60px;
  height: 60px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  -webkit-transition: ease-in-out 750ms;
  transition: ease-in-out 750ms;
  margin-right: 40px;
  box-shadow: rgba(50,50,93,0.25) 0px 2px 5px -1px, rgba(0,0,0,0.3) 0px 1px 3px -1px;
  cursor: pointer;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
}

.c8 p {
  font-size: 15px;
}

.c7 {
  display: none;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        마이페이지
      </div>
      <div
        class="c2"
        direction="horizontal"
        width="700px"
      />
    </div>
    <div>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          사진
        </div>
        <div
          class="c5"
        >
          <div>
            <label
              class="c6"
              for="img"
              style="background-image: url(https://user-images.githubusercontent.com/47776356/129712816-13701b24-57cc-451e-93c6-ca7afe190af1.jpeg);"
            />
            <input
              class="c7"
              id="img"
              type="file"
            />
          </div>
          <div
            class="c8"
          >
            <p>
              사진을 클릭하면 등록된 사진을 수정할 수 있습니다.
            </p>
            <p>
              등록된 사진은 상품 리뷰, 댓글 등에 사용됩니다.
            </p>
          </div>
        </div>
        <div />
      </div>
      <div
        class="c9"
        direction="horizontal"
        width="630px"
      />
      <div
        class="c10"
      >
        <div
          class="c11"
        >
          이메일
        </div>
        <input
          class="c12"
          disabled=""
          placeholder="수정은 수정 버튼으로"
          style="font-size: 14px;"
          type="email"
          value=""
          width="370px"
        />
        <div
          class="c13"
          disabled=""
        >
          수정
        </div>
      </div>
      <div
        class="c9"
        direction="horizontal"
        width="630px"
      />
      <div
        class="c10"
      >
        <div
          class="c11"
        >
          연락처
        </div>
        <div
          class="c14"
        >
          <input
            class="c15"
            disabled=""
            maxlength="3"
            placeholder="수정은"
            type="text"
            value=""
          />
          <img
            alt="hyphen"
            style="margin: 10px;"
          />
          <input
            class="c15"
            disabled=""
            maxlength="4"
            placeholder="수정"
            type="text"
            value=""
          />
          <img
            alt="hyphen"
            style="margin: 10px;"
          />
          <input
            class="c15"
            disabled=""
            maxlength="4"
            placeholder="버튼으로"
            type="text"
            value=""
          />
        </div>
        <div
          class="c13"
          disabled=""
        >
          수정
        </div>
      </div>
    </div>
  </div>
</div>
`;
